include/oneapi/tbb/concurrent_hash_map.h

```diff
@@ -467,9 +467,11 @@ class hash_map_iterator {
     friend class concurrent_hash_map;
 
     hash_map_iterator( const Container &map, std::size_t index, const bucket *b, node_base *n ) :
-        my_map(&map), my_index(index), my_bucket(b), my_node(static_cast<node*>(n))
+        my_map(&map), my_index(index), my_bucket(b), my_node(nullptr)
     {
-        if( b && !map_base::is_valid(n) )
+        if( map_base::is_valid(n) )
+            my_node = static_cast<node*>(n);
```

Comment:
Can you please add a comment in the code why we cannot do the cast during the initialization?

Reply:
Added.